### PR TITLE
geoserver: init at 2.21.0

### PR DESCRIPTION
**Pull request by sikmir**
###### Description of changes

> **[GeoServer](https://github.com/geoserver/geoserver)** is the reference implementation of the Open Geospatial Consortium (OGC) Web Feature Service (WFS) and Web Coverage Service (WCS) standards, as well as a high performance certified compliant Web Map Service (WMS), compliant Catalog Service for the Web (CSW) and implementing Web Processing Service (WPS). GeoServer forms a core component of the Geospatial Web.

[Packaging status](https://repology.org/project/geoserver/versions)

Try it out:
* `mkdir -p data_dir`
* `GEOSERVER_DATA_DIR=$PWD/data_dir result/bin/geoserver-startup`
* Go to http://localhost:8080/geoserver (admin:geoserver)
* Read more: https://docs.geoserver.org/latest/en/user/index.html

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
@GrahamcOfBorg eval

File: pkgs/servers/geospatial/geoserver/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchurl {
+    url = "mirror://sourceforge/geoserver/GeoServer/${version}/geoserver-${version}-bin.zip";
+    sha256 = "sha256-UCr22Ffhnux6eA0w5qoaf5Hvuypsl/FGpK+emi8G0Mc=";
+  };
+
+  sourceRoot = ".";
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/share/geoserver
```
Comment (legendofmiracles) [inline pkgs/servers/geospatial/geoserver/default.nix:16]: Please run the pre and postInstall phases

File: pkgs/servers/geospatial/geoserver/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  pname = "geoserver";
+  version = "2.21.0";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/geoserver/GeoServer/${version}/geoserver-${version}-bin.zip";
+    sha256 = "sha256-UCr22Ffhnux6eA0w5qoaf5Hvuypsl/FGpK+emi8G0Mc=";
+  };
+
+  sourceRoot = ".";
+  nativeBuildInputs = [ unzip makeWrapper ];
```
Comment (legendofmiracles) [inline pkgs/servers/geospatial/geoserver/default.nix:13]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```
I don't think that unzip is used

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -22930,6 +22908,32 @@ with pkgs;
 
   zipkin = callPackage ../servers/monitoring/zipkin { };
 
+  ### SERVERS / GEOSPATIAL
```
Comment (legendofmiracles) [inline pkgs/top-level/all-packages.nix:22911]: Please move the reformatting into a different commit

File: pkgs/servers/geospatial/geoserver/default.nix (hunk 2 above)
Comment (sikmir) [reply pkgs/servers/geospatial/geoserver/default.nix:13]: It's needed for unpacking, otherwise:
```
geoserver> do not know how to unpack source archive /nix/store/f9k2xn91f4zp6a7hsmr2kxpip57jk02v-geoserver-2.21.0-bin.zip
```
